### PR TITLE
SilverStripe 4 upgrade

### PR DESCRIPTION
**Pull request by jsirish**

### CODE REVIEW

**Comment by jsirish** [comment]
@muskie9 this is in pretty good shape. integrated a number of things from your PR as well. I think we should be able to merge and create issues as we run across them.

I did however have the image upload error on a fresh install from local composer after a 5.6 upgrade. Might be worth a test against a fresh install of alpha7

File: src/FlexSlider.php [hunk 1]
```diff
@@ -59,11 +88,7 @@ public function updateCMSFields(FieldList $fields)
         // Slides
         if ($this->owner->ID) {
             $config = GridFieldConfig_RecordEditor::create();
-            $config->addComponent(new GridFieldOrderableRows('SortOrder'));
-            if (class_exists('GridFieldBulkUpload')) {
-                $config->addComponent(new GridFieldBulkUpload());
-                $config->addComponent(new GridFieldBulkManager());
-            }
+            //$config->addComponent(new GridFieldOrderableRows('SortOrder'));
```
Comment (muskie9) [inline src/FlexSlider.php]: With gridfieldextensions having a 4 compatible branch now, does orderable work?

**Comment by muskie9** [comment]
Should we update calls such as `$before = (method_exists($this->owner->ClassName, 'flexSliderBeforeAction'))` to something like `$before = $this->owner->hasMethod('flexSliderBeforeAction')`? I'm not sure if there's a performance benefit to either.

File: src/FlexSlider.php (hunk 1 above)
Comment (jsirish) [inline src/FlexSlider.php]: did a composer update, was an update for extensions - works now!

**Comment by jsirish** [comment]
I've updated the JS call, and switched in `hasMethod` for `method_exists`. 

Verified it's working on the front end, though you need to add the `<% include FlexSlider %>` to Page.ss in Simple theme, and remove the two JS calls in the top level Page.ss. Thumbnail nav works as well.

All tests pass locally and on Travis other than SQLite.

**Comment by muskie9** [comment]
This checked out locally, I've restarted tests to see if the sqlite will pass. The only thing I ran into is the Simple theme jquery inclusion caused the frontend to fail. Removing that inclusion from the main `Page.ss` file fixed it.

**Pull request closed without merging**